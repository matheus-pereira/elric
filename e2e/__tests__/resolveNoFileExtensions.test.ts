/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {wrap} from 'elric-snapshot-serializer-raw';
import {cleanup, extractSummary, writeFiles} from '../Utils';
import runelric from '../runelric';

const DIR = path.resolve(__dirname, '../resolve-no-extensions-no-js');

beforeEach(() => cleanup(DIR));
afterAll(() => cleanup(DIR));

test('show error message with matching files', () => {
  const {exitCode, stderr} = runelric('resolve-no-extensions');
  const {rest} = extractSummary(stderr);

  expect(exitCode).toBe(1);
  expect(wrap(rest)).toMatchSnapshot();
});

test('show error message when no js moduleFileExtensions', () => {
  writeFiles(DIR, {
    'index.jsx': `
      module.exports ={found: true};
    `,
    'package.json': `
      {
        "elric": {
          "moduleFileExtensions": ["jsx"]
        }
      }
    `,
    'test.jsx': `
      const m = require('../');

      test('some test', () => {
        expect(m.found).toBe(true);
      });
    `,
  });

  const {exitCode, stderr} = runelric('resolve-no-extensions-no-js');

  expect(exitCode).toBe(1);
  expect(wrap(stderr)).toMatchSnapshot();
});
