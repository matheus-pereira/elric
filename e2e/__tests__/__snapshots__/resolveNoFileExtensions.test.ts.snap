// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`show error message when no js moduleFileExtensions 1`] = `
● Validation Error:

  moduleFileExtensions must include 'js':
  but instead received:
    ["jsx"]
  Please change your configuration to include 'js'.

  Configuration Documentation:
  https://elricjs.io/docs/configuration

`;

exports[`show error message with matching files 1`] = `
FAIL __tests__/test.js
  ● Test suite failed to run

    Cannot find module './some-json-file' from 'index.js'

    Require stack:
      index.js
      __tests__/test.js


    However, elric was able to find:
    	'./some-json-file.json'

    You might want to include a file extension in your import, or update your 'moduleFileExtensions', which is currently ['js'].

    See https://elricjs.io/docs/configuration#modulefileextensions-arraystring

      6 |  */
      7 |
    > 8 | module.exports = require('./some-json-file');
        |                  ^
      9 |

      at Resolver.resolveModule (../../packages/elric-resolve/build/resolver.js:324:11)
      at Object.require (index.js:8:18)
`;
